src/targets/gpu/device_name.cpp

```diff
@@ -52,13 +52,15 @@ std::string get_device_name()
 bool gfx_has_fp8fnuz_intrinsics()
 {
     const auto device_name = trim(split_string(get_device_name(), ':').front());
-    return (starts_with(device_name, "gfx9") and device_name >= "gfx940");
+    return (starts_with(device_name, "gfx94"));
```

Comment:
How do you deal with MI100, MI2xx ?

Reply:
They don't officially support fp8. They weren't going through with the previous condition anyways.